pkg/openstack/loadbalancer.go

```diff
@@ -2248,16 +2243,23 @@ func (lbaas *LbaasV2) ensureAndUpdateOctaviaSecurityGroup(clusterName string, ap
 	}
 
 	mc := metrics.NewMetricContext("subnet", "get")
-	subnet, err := subnets.Get(lbaas.network, memberSubnetID).Extract()
+	subnet, err := subnets.Get(lbaas.network, svcConf.lbMemberSubnetID).Extract()
 	if mc.ObserveRequest(err) != nil {
 		return fmt.Errorf(
-			"failed to find subnet %s from openstack: %v", memberSubnetID, err)
+			"failed to find subnet %s from openstack: %v", svcConf.lbMemberSubnetID, err)
 	}
 
 	etherType := rules.EtherType4
 	if netutils.IsIPv6CIDRString(subnet.CIDR) {
 		etherType = rules.EtherType6
 	}
+	cidrs := []string{subnet.CIDR}
+	if lbaas.opts.LBProvider == "ovn" {
```

Comment:
Do I vaguely recall a previous discussion about providers having arbitrary names? I don't remember who brought it up: either you or @kayrus I'm guessing?

Reply:
Yes. Later on I consulted Octavia team to learn I was wrong:

> the providers should be explicitly listed in the octavia config file ([http://rhos-ci-logs.lab.eng.tlv2.redhat.com/logs/rcj/DFG-network-octavia-17.1_director-rhel-vi[…]ppet-generated/octavia/etc/octavia/octavia.conf.gz](http://rhos-ci-logs.lab.eng.tlv2.redhat.com/logs/rcj/DFG-network-octavia-17.1_director-rhel-virthost-3cont_2comp_1ipa-ipv4-geneve-tls/34/controller-0/var/lib/config-data/puppet-generated/octavia/etc/octavia/octavia.conf.gz))
> ```
> # Dictionary of enabled provider driver names and descriptions
> # A comma separated list of dictionaries of the enabled provider driver names
> # and descriptions.
> # enabled_provider_drivers = amphora:The Octavia Amphora driver.,octavia: \
> enabled_provider_drivers=amphora: The Octavia Amphora driver.,octavia: Deprecated alias of the Octavia Amphora driver.,ovn: Octavia OVN driver.
> #                            Deprecated alias of the Octavia Amphora driver.
> ```
> then ovn-provider defines an "official" name for its plugin: https://opendev.org/openstack/ovn-octavia-provider/src/branch/master/setup.cfg#L30-L31